clippy_lints/src/utils/mod.rs

```diff
@@ -363,7 +418,7 @@ pub fn implements_trait<'tcx>(
     if ty.has_infer_types() {
         return false;
     }
-    let ty = cx.tcx.erase_regions(&ty);
+    let ty = cx.tcx.erase_regions(ty);
```

Comment:
Unrelated changes. Probably needs a rebase.

Reply:
Yeah, this one needs a rebase